### ISSUE TITLE
Broken Spree::Store migration with Globalize

### ISSUE THREAD

**Issue by dangerdogz** [solidusio/solidus#485]
I am currently working on the SolidusGlobalize port, and I have the following issue with a specific core migrations (see: [relevant file](https://github.com/solidusio/solidus/blob/master/core/db/migrate/20140309033438_create_store_from_preferences.rb))

```ruby
class CreateStoreFromPreferences < ActiveRecord::Migration
  def change
    preference_store = Spree::Preferences::Store.instance
    if store = Spree::Store.where(default: true).first
      store.meta_description = preference_store.get('spree/app_configuration/default_meta_description') {}
      store.meta_keywords    = preference_store.get('spree/app_configuration/default_meta_keywords') {}
      store.seo_title        = preference_store.get('spree/app_configuration/default_seo_title') {}
      store.save!
    else
      # we set defaults for the things we now require
      Spree::Store.new do |s|
        s.name              = preference_store.get 'spree/app_configuration/site_name' do
          'Sample Store'
        end
        s.url               = preference_store.get 'spree/app_configuration/site_url' do
          'example.com'
        end
        s.mail_from_address = preference_store.get 'spree/app_configuration/mails_from' do
          'store@example.com'
        end

        s.meta_description = preference_store.get('spree/app_configuration/default_meta_description') {}
        s.meta_keywords    = preference_store.get('spree/app_configuration/default_meta_keywords') {}
        s.seo_title        = preference_store.get('spree/app_configuration/default_seo_title') {}
        s.default_currency = preference_store.get('spree/app_configuration/currency') {}
        s.code             = 'spree'
      end.save!
    end
  end
end
```

When you add translation support to the Store model with Globalize, there is an error because the migration initialize and save a record before the spree_store_translations table can be created, so since the translations table doesn't exist yet, the save! fails.

[Stacktrace](https://gist.github.com/dangerdogz/be016f3d3088c8e120a0)

The issue was filed previously in Spree, see [Spree#6259 file](https://github.com/spree/spree/pull/6259)

It was fixed by using class_eval on Spree::Store to disable the translation features for the scope of the migration

I'm opening the discussion before offering a solution, migrations is a somewhat sensitive part and I'm not familiar enough with the project and the code standards to know if the above solution is good enough or if we could defer the creation of the default Store and move it somewhere else where it's more obvious, but I'm looking into it and will provide a pull request and need test coverage if someone can point me in the proper direction

**Comment by athal7** [comment]
@dangerdogz when does the table usually get created?

**Comment by dangerdogz** [comment]
In a migration added by Solidus/SpreeGlobalize so outside the core itself

It's not a problem for other translated models since no records is created between the Solidus Core migrations and the Solidus Globalize migrations in theory

It's only a problem for new stores / bundle exec test_app task

**Comment by athal7** [comment]
@dangerdogz can you link to the place where that table is usually created please

**Comment by dangerdogz** [comment]
[This migration from SpreeGlobalize](https://github.com/spree-contrib/spree_globalize/blob/master/db/migrate/20150323210949_add_translations_to_store.rb)

**Comment by athal7** [comment]
per offline chat with @dangerdogz, we think a good path forward here would be to move the store creation to the seeds file, then the migrations would go fine, and db:schema:load could also be used.

@jhawthorn curious if you have more context here and whether that will cause any issues

**Comment by jhawthorn** [comment]
I believe we should create a store during migrations. Someone upgrading an existing site should have a `Store` created for them during migrations.

This is a prime example of why we have to avoid using AR models in migrations: later changes to their behaviour will break older migrations.

**Comment by dangerdogz** [comment]
Fixed by #488
